languages/rust/oso/src/host/class.rs

```diff
@@ -231,8 +269,9 @@ where
         Args: FromPolarList,
```

Comment:
Should we remove `add_iterator_method`?  We no longer allow this multiple result behavior without `in`, right?

Reply:
Other question, should we remove the ability for single calls to return multiple results without the use of the `NextExternal` event.

Reply:
Not sure I follow, here's what's happening here:

`add_iterator_method` ensures that the result is iterable, and wraps it in the `PolarIterator` struct. This struct acts as a registered class, and `in` works on this class.

(Mid-sentence realised this could be better. Will make it so that `PolarIterator` also implements `as_iter`, so that we can remove the manual downcasting.)

It's already true that multiple results can only be retrieved using `in`. You can do `x = foo.iter_method()` to get back the PolarIterator, but all you can do with this class is basically `y in x or z in x`.